### PR TITLE
Update docs and remove all warnings [skip-ci]

### PR DESCRIPTION
**Pull request by ayushdg**
This pr fixes the following 
- Add traveling salesperson problem to the docs
- Update docs to address all build warnings

To remove some warnings. updated the use of `NOTE:` in cases like the one shown below.

| Old  | New |
| ------------- | ------------- |
| image | image  |

### CODE REVIEW

**Review by BradReesWork — APPROVED** [review]

**Comment by BradReesWork** [comment]
@ayushdg I would like to push this into branch 0.19

**Review by rlratzel — APPROVED** [review]
Approving contingent on resolving the issue about `minimum_spanning_tree` under "Maximum Spanning Tree" in `api.rst` (either changing or ignoring if it's correct as-is).

**Review by ajschmidt8 — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

File: docs/source/api.rst [hunk 1]
```diff
@@ -264,27 +271,25 @@ Minimum Spanning Tree
...
-    :members:
+    :members: minimum_spanning_tree
     :undoc-members:
 
 Maximum Spanning Tree
 ---------------------
 
-.. automodule:: cugraph.tree.maximum_spanning_tree
-    :members:
+.. automodule:: cugraph.tree.minimum_spanning_tree
```
[the remark this answers is not in the document]
Comment (ayushdg) [reply docs/source/api.rst:280]: Yup, the file `minimum_spanning_tree.py` contains the methods, `minimum_spanning tree` and `maximum_spanning_tree`. Updated docs to include one member of the module, per header.

**Pull request merged**